src/js/services/integrations-service.js

```diff
@@ -37,17 +38,55 @@ export function resetTwilioAttributes() {
 export function fetchTwilioAttributes() {
     const {user: {clients, pendingClients}} = store.getState();
     const client = clients.find((client) => client.platform === 'twilio');
-    const pendingClient = pendingClients.find((client) => client.platform === 'twilio');
+    const pendingClient = pendingClients && pendingClients.find((client) => client.platform === 'twilio');
 
     if (client) {
         updateTwilioAttributes({
             linkState: 'linked',
-            number: client.displayName
+            appUserNumber: client.displayName
         });
     } else if (pendingClient) {
         updateTwilioAttributes({
             linkState: 'pending',
-            number: pendingClient.displayName
+            appUserNumber: pendingClient.displayName
         });
     }
 }
+
+export function linkTwilioChannel(userId, data) {
+    return core().appUsers.link.linkChannel(userId, data)
+        .then((appUser) => {
+            immediateUpdate(appUser).then(() => {
+                updateTwilioAttributes({
+                    linkState: 'pending'
+                });
+            });
+        })
+        .catch(() => {
+            updateTwilioAttributes({
+                hasError: true,
+                errorMessage: 'Something went wrong. Please try again.'
+            });
+        });
+}
+
+export function deleteTwilioChannel(userId) {
+    return core().appUsers.link.deleteTwilioChannel(userId)
+        .then(() => {
+            immediateUpdate({
+                pendingClients: []
```

Comment:
If the server returns the appUser on link, I would expect it to return it on unlink too. Also, it's not possible to have more than one *twilio* pending client, but what happens when we have pending email client too?